// @krsbx/prisma-repo/dist/utils/constants.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PKG_ROOT = exports.PRISMA_LOGGER_VALUES = exports.PRISMA_LOGGER = exports.TYPE_SUFFIX = exports.GENERATOR_NAME = void 0;
const path_1 = __importDefault(require("path"));
exports.GENERATOR_NAME = 'Prisma Repo Generator';
exports.TYPE_SUFFIX = {
    WHERE_INPUT: 'WhereInput',
    SELECT: 'Select',
    INCLUDE: 'Include',
    CREATE_INPUT: 'CreateInput',
    UPDATE_INPUT: 'UpdateInput',
    WHERE_UNIQUE_INPUT: 'WhereUniqueInput',
    ORDER_BY_WITH_RELATION_INPUT: 'OrderByWithRelationInput',
    DELEGATE: 'Delegate',
    GROUP_BY: 'GroupByOutputType',
    RETURN: 'Return',
};
exports.PRISMA_LOGGER = {
    QUERY: 'query',
    INFO: 'info',
    WARN: 'warn',
    ERROR: 'error',
};
exports.PRISMA_LOGGER_VALUES = Object.values(exports.PRISMA_LOGGER);
exports.PKG_ROOT = path_1.default.join(__dirname, '../../');
